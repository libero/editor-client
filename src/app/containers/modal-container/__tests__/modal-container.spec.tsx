import React from 'react';
import { create } from 'react-test-renderer';
import { ModalContainer } from '../index';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

jest.mock('@material-ui/core', () => ({
  Dialog: ({ children }) => <div data-cmp="dialog">{children}</div>,
  DialogTitle: ({ children }) => <div data-cmp="dialog-title">{children}</div>,
  DialogActions: ({ children }) => <div data-cmp="dialog-actions">{children}</div>,
  DialogContent: ({ children }) => <div data-cmp="dialog-content">{children}</div>,
  DialogContentText: ({ children }) => <div data-cmp="dialog-content-text">{children}</div>,
  Content: ({ children }) => <div data-cmp="dialog-content">{children}</div>,
  Typography: () => <div data-cmp="typography"></div>,
  Button: () => <div data-cmp="Button"></div>
}));

describe('Modal Container', () => {
  const mockStore = configureMockStore([]);

  it('renders modal', () => {
    const store = mockStore({
      manuscriptEditor: {
        modal: {
          params: {
            component: () => <div data-cmp="test-component" />
          },
          isVisible: true
        }
      }
    });

    const wrapper = create(
      <Provider store={store}>
        <ModalContainer />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
