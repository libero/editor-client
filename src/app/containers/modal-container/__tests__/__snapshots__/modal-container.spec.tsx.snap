// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Container renders modal 1`] = `
<div
  data-cmp="dialog"
>
  <div
    data-cmp="dialog-title"
  >
    <div
      data-cmp="typography"
    />
  </div>
  <div
    data-cmp="dialog-content"
  >
    <div
      data-cmp="test-component"
    />
  </div>
</div>
`;
